src/main/java/com/commercetools/emailprocessor/email/EmailProcessor.java

```diff
@@ -63,42 +61,22 @@ public EmailProcessor() {
      */
     public CompletionStage<Statistics> processEmails(final TenantConfiguration tenantConfig) {
         SphereClient client = tenantConfig.getSphereClient();
-        CustomObjectQuery<JsonNode> query = CustomObjectQuery.ofJsonNode().byContainer(CONTAINER_ID).withLimit(100L)
-            .withSort(s -> s.createdAt().sort().asc());
-        return client
-            .execute(query)
-            .thenApply(response -> {
-                Statistics statistics = new Statistics(tenantConfig.getProjectKey());
-                if (response.getTotal() < 1) {
-                    LOG.info(String.format("No email to process for tenant %s", tenantConfig
-                        .getProjectKey()));
-                }
-                List<CompletableFuture<Void>> allTenants = response.getResults().stream()
-                    .map(customObject -> Optional.ofNullable(customObject)
-                        .map(CustomObject::getValue)
-                        .map(node -> node.get(EMAIL_PROPERTY_STATUS))
-                        .map(JsonNode::asText)
-                        .filter(status -> ((equalsIgnoreCase(status, STATUS_PENDING) || tenantConfig.isProcessAll())))
-                        .map(pending -> callApiEndpoint(customObject.getId(), tenantConfig)
-                            .thenAccept(statistics::update))
-                        .orElseGet(() -> {
-                            statistics.update(Statistics.RESPONSE_IGNORED);
-                            return CompletableFuture.completedFuture(null);
-                        }))
-                    .collect(toList());
+        CustomObjectQuery<JsonNode> query = CustomObjectQuery.ofJsonNode().byContainer(CONTAINER_ID);
+        if (!tenantConfig.isProcessAll()) {
+            query = query.plusPredicates(QueryPredicate.of("value(status=\"" + STATUS_PENDING + "\")"));
+        }
+        //sorting the emailobject by creationt ime, to ensure that the emails are delivered in the correct chronically
```

Comment:
@ahalberkamp Some typos. Maybe ->

`// We sort the email object by creation time, to ensure that the emails are delivered in the correct chronological order they were sent with`

Reply:
7c4c7f4e01e02c348f5b58790a081fd42e252ece